### PR TITLE
chore: Update readme with new language

### PR DESCRIPTION
**Pull request by binarylogic**
* Updates the readme to reflect our new marketing language.
* Drops the `README.md.erb` since we are no longer generating files in the Vector repo.
* Adds a comparison matrix.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,45 +14,32 @@
...
 ## What is Vector?
 
-Vector is a lightweight, ultra-fast, [open-source][urls.vector_repo] tool for
-building observability pipelines. Compared to Logstash and friends, Vector
-[improves throughput by ~10X while significantly reducing CPU and memory
-usage][urls.vector_performance].
+Vector is a high-performance observability data platform that puts you in
+control of your observability data. [Collect][docs.sources],
+[transform][docs.transforms], and [route][docs.sinks] all your logs, metrics,
+and traces to any vendors you want today and any other vendors you want
```
Comment (jamtur01) [inline README.md:22]: you may want ...

File: README.md [hunk 2]
```diff
@@ -10,45 +14,32 @@
...
+tomorrow. Vector enables dramatic cost reduction, novel data enrichment, and
+data security where you need it, not where is most convenient for your vendors.
```
Comment (jamtur01) [inline README.md:24]: Not sure "where" works here? When perhaps?

File: README.md [hunk 3]
```diff
@@ -10,45 +14,32 @@
...
-* You _SHOULD_ use Vector to replace Logstash, Fluent*, Telegraf, Beats, or similar tools.
-* You _SHOULD_ use Vector as a [daemon][docs.strategies#daemon] or [sidecar][docs.strategies#sidecar].
-* You _SHOULD_ use Vector as a Kafka consumer/producer for observability data.
-* You _SHOULD_ use Vector in resource constrained environments (such as devices).
-* You _SHOULD NOT_ use Vector if you need an advanced distributed stream processing framework.
-* You _SHOULD NOT_ use Vector to replace Kafka. Vector is designed to work with Kafka!
-* You _SHOULD NOT_ use Vector for non-observability data such as analytics data.
+* **Reliability** - Built in [Rust][urls.rust], Vector's primary design goal is reliability & performance.
+* **End-to-end** - Deploys as an agent or aggregator. One tool. One workflow.
+* **All data** - [Logs][docs.data-model.log], [metrics][docs.data-model.metric], and traces (coming soon). One tool. All data.
```
Comment (jamtur01) [inline README.md:32]: All your data?

**Comment by binarylogic** [comment]
@jamtur01 addressed all feedback. Gimme that check!

**Review by jamtur01 — APPROVED** [review]
LGTM

giphub

**Pull request merged**